ArjunaJTA/jdbc/classes/com/arjuna/ats/internal/jdbc/ConnectionImple.java

```diff
@@ -379,7 +379,8 @@ public void close() throws SQLException
 
 	        if (!delayClose)  // close now
 	        {
-	            _transactionalDriverXAConnectionConnection.closeCloseCurrentConnection();
+	            if (_theModifier != null)
```

Comment:
So if the driver is not in the modifier list then you do not register an AC to close the connection. Similarly, at this line, if its not in the modifier list then you do not call close on the TransactionalDriverXAConnection connection. But, as I recall, the original bug was that TransactionalDriverXAConnection#close() was not getting called so I would say the fix is not sufficient.